deps/src/singular.cpp

```diff
@@ -53,6 +74,52 @@ JLCXX_MODULE define_julia_module(jlcxx::Module & Singular)
     singular_define_ideals(Singular);
     singular_define_matrices(Singular);
 
+
+    // Calls the Singular interpreter with `input`.
+    // `input` needs to be valid Singular input. 
+    // Returns a 4-tuple:
+    // 1. entry is a bool, indicated if an error has happened
+    // 2. entry is the output as a string
+    // 3. entry is the error output as a string
+    // 4. entry is the warning output as a string
+    Singular.method("call_interpreter", [](std::string input) {
+        // save callbacks
+        auto default_print = PrintS_callback;
+        auto default_error = WerrorS_callback;
+        auto default_warning = WarnS_callback;
+
+        // set temporary new callbacks
+        PrintS_callback = PrintS_for_julia;
+        WerrorS_callback = WerrorS_for_julia;
+        WarnS_callback = WarningS_for_julia;
+
+        // cleanup return strings
+        singular_return.clear();
+        singular_error.clear();
+        singular_warning.clear();
+
+        // call interpreter
+        std::string input_str = input + "\nreturn();";
+        bool err = iiAllStart(NULL, const_cast<char *>(input_str.c_str()),
```

Comment:
I made https://github.com/Singular/Sources/pull/931 so we could simplify this in the future.